generator/restriction_collector.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "routing/routing_serializer.hpp"
```

Comment:
Please change the file name if it is not too hard. Compare with existing:
base/std_serialization.hpp
indexer/geometry_serialization.hpp